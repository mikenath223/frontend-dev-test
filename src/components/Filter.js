
const Filter = ({ entry, handleInputChange }) => {
  return (<input type="text" value={entry} onChange={handleInputChange} />)
}

export default Filter;