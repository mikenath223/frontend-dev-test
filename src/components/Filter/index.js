import PropTypes from 'prop-types';
import styles from './Filter.module.css';

const Filter = ({ handleInputChange }) => (
  <input
    className={styles.search}
    type="text"
    onChange={handleInputChange}
    placeholder="Search..."
  />
);

Filter.propTypes = {
  handleInputChange: PropTypes.func.isRequired,
};

export default Filter;
