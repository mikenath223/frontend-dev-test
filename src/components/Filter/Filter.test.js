import { render } from '@testing-library/react';
import Filter from './index';

describe('Filter', () => {
  it('should match snapshot', () => {
    const { container } = render(<Filter handleInputChange={jest.fn()} />);

    expect(container).toMatchSnapshot();
  });
});
