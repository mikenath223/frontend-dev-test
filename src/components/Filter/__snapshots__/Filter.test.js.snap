// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter should match snapshot 1`] = `
<div>
  <input
    class="search"
    placeholder="Search..."
    type="text"
  />
</div>
`;
